src/renci_ner/services/linkers/nameres.py

```diff
@@ -0,0 +1,118 @@
+            "only_prefixes": "(list of prefixes, default: []) The prefixes to filter results to, combined with OR.",
+            "exclude_prefixes": "(list of prefixes, default: []) The prefixes to exclude from search results, combined with AND.",
+            "only_taxa": "(list of taxa, default: []) The taxa to filter results to as NCBITaxon identifiers, combined with OR.",
+        }
+
+    def annotate(self, text, props=None) -> AnnotatedText:
+        """
+        Annotate a piece of text using NameRes.
+
+        :param text: A piece of text with the label of a biomedical entity (e.g. "brain" or "ACT1").
+        :param props: A dictionary of properties to configure NameRes.
+        :return: An AnnotatedText object containing the annotations.
+        """
+        if props is None:
+            props = {}
+        session = self.requests_session
+
+        response = session.get(
+            self.lookup_url,
+            params={
+                "string": text,
+                "autocomplete": props.get("autocomplete", "false"),
+                "limit": props.get("limit", 10),
+                "highlighting": props.get("highlighting", "false"),
+                "biolink_type": "|".join(props.get("biolink_types", [])),
+                "only_prefixes": "|".join(props.get("only_prefixes", [])),
+                "exclude_prefixes": "|".join(props.get("exclude_prefixes", [])),
+                "only_taxa": "|".join(props.get("only_taxa", [])),
+            },
+        )
+
+        response.raise_for_status()
+        results = response.json()
+
+        annotations = []
+        for result in results:
+            annotations.append(
+                # Since NameRes is normalized to Babel, we can treat it as a NormalizedAnnotation.
+                NormalizedAnnotation(
+                    text=text,
+                    id=result.get("curie", ""),
+                    label=result.get("label", ""),
+                    biolink_type=result.get("types", ["biolink:NamedThing"])[0],
+                    type=result.get("types", ["biolink:NamedThing"])[0],
+                    props={
+                        "score": result.get("score", 0),
+                        "clique_identifier_count": result.get(
+                            "clique_identifier_count", 0
+                        ),
+                        "synonyms": result.get("synonyms", []),
+                        "highlighting": result.get("highlighting", {}),
+                        "types": result.get("types", []),
+                        "taxa": result.get("taxa", []),
+                    },
+                    provenance=self.provenance,
+                    # Since we're using the whole text, let's just use that
+                    # as the start/end.
+                    start=0,
+                    end=len(text),
+                )
+            )
```

Comment:
It'd be more performant to replace the loop with a list comprehension, something like ```annotations=[NormalizedAnnotation(text=text, id=result.get("curie", ""), ....) for result in results]```

Reply:
Fixed in 4dc6ee3.